rest/blip_sync.go

```diff
@@ -503,26 +526,28 @@ func (bh *blipHandler) sendRevision(sender *blip.Sender, seq db.SequenceID, docI
 // Received a "rev" request, i.e. client is pushing a revision body
 func (bh *blipHandler) handleAddRevision(rq *blip.Message) error {
 
+	addRevision := newAddRevision(rq)
+	bh.logEndpointEntry(rq.Profile(), addRevision)
+
 	var body db.Body
 	if err := rq.ReadJSONBody(&body); err != nil {
 		return err
 	}
 
 	// Doc metadata comes from the BLIP message metadata, not magic document properties:
-	docID, found := rq.Properties["id"]
-	revID, rfound := rq.Properties["rev"]
+	docID, found := addRevision.id()
+	revID, rfound := addRevision.rev()
 	if !found || !rfound {
 		return base.HTTPErrorf(http.StatusBadRequest, "Missing docID or revID")
 	}
-	if del, found := rq.Properties["deleted"]; found && del != "0" && del != "false" {
+	if del, found := addRevision.deleted(); found && del != "0" && del != "false" {
```

Comment:
I feel like we'd be making progress if addRevision.deleted() was getting rid of code like "found && del != "0" && del != "false"

Reply:
I moved that logic into `addRevision.deleted()` and added isolated test coverage: https://github.com/couchbase/sync_gateway/pull/3246/commits/367f92888ff98b5bb98c22dbafbc9048cda26eeb

That's the primary motivation of trying to refactor these helpers out -- to be able to have isolated unit tests as they increase in complexity or edge case bugs arise.

Reply:
> That's the primary motivation of trying to refactor these helpers out -- to be able to have isolated unit tests as they increase in complexity or edge case bugs arise.

Actually I take that back .. the primary motivation was to be able to concisely emit the parameters of the request in single `String()` call.